chitchat/src/listener.rs

```diff
@@ -0,0 +1,246 @@
+use std::collections::{BTreeMap, HashMap};
+use std::ops::Bound;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::{Arc, RwLock, Weak};
+
+pub struct ListenerHandle {
+    prefix: String,
+    idx: usize,
```

Comment:
I haven't read the rest of the file yet, so I have no idea what that means. Is that the handle ID or some index into something else? A better name/comment would help.